gh-squash-merge

```diff
@@ -37,30 +45,73 @@ but generally we tend towards
 <!-- SQUASH_MERGE_END -->
 
 EXAMPLES
+gh squash-merge
 gh squash-merge 811
+gh squash-merge --repo quotidian-ennui/gh-squash-merge 811
+gh squash-merge https://github.com/quotidian-ennui/gh-squash-merge/pull/811
+gh squash-merge feature/update
 
 EOF
- exit 1
+  exit 1
 }
 
 write_commit_msg() {
   local gh_repo="$1"
   local pr_number="$2"
-  gh_api "repos/$gh_repo/pulls/$pr_number" -q ".body" | awk '/SQUASH_MERGE_START/,/SQUASH_MERGE_END/' | grep -v "SQUASH_MERGE" > "$WORK_FILE"
+  gh_api "repos/$gh_repo/pulls/$pr_number" -q ".body" | awk '/SQUASH_MERGE_START/,/SQUASH_MERGE_END/' | grep -v "SQUASH_MERGE" >"$WORK_FILE"
   echo "$WORK_FILE"
 }
 
 trap cleanup 1 2 15
 
-pr_number="$1"
-gh_repo=$(gh repo view --json "nameWithOwner" -q ".nameWithOwner") || true
-
-if [[ -z "${gh_repo:-}" || -z "${pr_number:-}" ]]; then
-  echo "Sorry, need to be in a github repo and have a PR number"
+ARGS=$(getopt --options 'R::h' --longoptions 'repo:,help' -- "${@}")
+eval "set -- ${ARGS}"
+while true; do
+  case "${1}" in
+  --h | --help)
+    usage
+    ;;
+  -R | --repo)
+    repo="${2}"
+    shift 2
+    ;;
+  --)
+    shift
+    break
+    ;;
+  *)
+    usage
+    ;;
+  esac
+done
+
+remaining_args=("${1}")
+
+if [ -n "${repo:-}" ] && [ -z "${remaining_args:-}" ]; then
+  echo "argument required when using the --repo flag"
   usage
 fi
 
+details_args=()
+
+if [ -n "${repo:-}" ]; then
+  details_args+=('--repo')
+  details_args+=("${repo}")
+fi
+
+if [ -n "${remaining_args:-}" ]; then
+  details_args+=("${remaining_args}")
+fi
+
+pr_details=$(gh pr view --json headRepository,headRepositoryOwner,number $"${details_args[@]}")
```

Comment:
This doesn't work as it returns where its coming from which in this case is a fork.

Reply:
it always worked in my other use case because it was always run in a  repo i had created the branch in.

Reply:
It seems to me that we just need to not use `gh api` because the detailed_args gives us everything we need for interacting with `gh pr`

```
write_commit_msg() {
...
  gh pr view $detailed_args --json "body" -q ".body" | awk '/SQUASH_MERGE_START/,/SQUASH_MERGE_END/' | grep -v "SQUASH_MERGE" > "$WORK_FILE"
  echo "$WORK_FILE"
}

title=$(gh pr view $detailed_args --json "title" -q ".title")
....
```

Reply:
Would you prefer a dependency on `jq` or a multiple API calls?

```bash
pr_details=$(gh pr view --json headRepository,headRepositoryOwner,number $"${details_args[@]}")

title=$(jq -r .title <<<"$pr_details")
body=$(jq -r .body <<<"$pr_details")

commit_msg_file=$(write_commit_msg "$body")
```

vs

```bash
write_commit_msg() {
//...
  gh pr view $detailed_args --json "body" -q ".body" | awk '/SQUASH_MERGE_START/,/SQUASH_MERGE_END/' | grep -v "SQUASH_MERGE" > "$WORK_FILE"
  echo "$WORK_FILE"
}

title=$(gh pr view $detailed_args --json "title" -q ".title")
//....
```

I'm leaning towards the latter.

Reply:
I'm leaning towards the former; purely because using `gh api` is an extra API call. Since we already need 'awk' reducing the number of external tools is somewhat a moot point?

Also, corporate laptops and the scanning of app executions...